apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/(analysis)/summary/components/SurveyAnalysisCTA.test.tsx

```
@@ -766,4 +819,224 @@ describe("SurveyAnalysisCTA", () => {
+    // For read-only users, there should be no icon bar actions
+    expect(screen.queryAllByTestId(/icon-bar-action-/)).toHaveLength(0);
+  });
+
+  test("opens reset confirmation modal when reset button is clicked", async () => {
+    const user = userEvent.setup();
+    render(<SurveyAnalysisCTA {...defaultProps} responseCount={5} />);
+
+    const iconActions = screen.getAllByTestId(/icon-bar-action-/);
+    const resetButton = iconActions.find((button) => button.getAttribute("title") === "Reset survey");
+
+    await user.click(resetButton!);
```

Biome (1.9.4)

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)